rero_ils/modules/loans/api.py

```diff
@@ -822,6 +872,40 @@ def loan_has_open_events(loan_pid=None):
     return False
 
 
+def get_non_anonymized_loans(patron_pid=None):
+    """Search all loans for non anonymized loans.
+
+    :param patron_pid: optional parameter to filter by patron_pid.
+    :return: loans.
+    """
+    loans = []
+    search = current_circulation.loan_search_cls()\
+        .filter('term', to_anonymize=False)\
+        .source(['pid'])
+    if patron_pid:
+        search = search.filter('term', patron__pid=patron_pid)
+    for record in search.scan():
+        loans.append(Loan.get_record_by_pid(record.pid))
+    return loans
```

Comment:
This can return a huge number of loans (several thousand !). Isn't exists a better way to find loans to anonymized ? (loop for patron asking to anonymized + loop on loans link to this patron not anonymized) ?
At least use a `yield` statement to avoid memory overflow.

Reply:
Filter by state is now in place + execution by organisation + yied...